src/ir/from_ast/astconv.rs

```diff
@@ -855,5 +975,8 @@ pub fn transform(ns: ast::Namespace) -> ir::Context {
         let ir_comp = BuildCtx::comp(comp, idx, &sig_map);
         ctx.comps.checked_add(idx, ir_comp);
     }
+
+    connect(&mut ctx);
```

Comment:
Hmm, I'm a bit confused about this. I thought that this is operating on the `ast::Connect` nodes but it seems to do something completely different. Can you explain what this method is supposed to do and why it is running after the all the components have been transformed?